File: java/leetcode_grind/Day891.java
package leetcode_grind;

public class Day891 {
    // https://leetcode.com/problems/count-subarrays-with-score-less-than-k/description/?envType=daily-question&envId=2025-04-28
    static class Solution1 {
        public long countSubarrays(int[] nums, long k) {
            int n = nums.length;
            long res = 0, total = 0;
            for (int i = 0, j = 0; j < n; j++) {
                total += nums[j];
                while (i <= j && total * (j - i + 1) >= k) {
                    total -= nums[i];
                    i++;
                }
                res += j - i + 1;
            }
            return res;
        }
    }
}

File: src/leetcode_grind/day_891.rs
// https://leetcode.com/problems/count-subarrays-with-score-less-than-k/description/?envType=daily-question&envId=2025-04-28
pub fn count_subarrays(nums: Vec<i32>, k: i64) -> i64 {
    let n = nums.len();        
    let mut res = 0;
    let mut total = 0;
    let mut i = 0;
    for j in 0..n {
        total += nums[j] as i64;
        while i <= j && total * (j - i + 1) as i64 >= k {
            total -= nums[i] as i64;
            i += 1;
        }
        res += j - i + 1;
    }
    res as i64
}